warehouse/query-core/src/main/java/datawave/query/iterator/QueryIterator.java

```diff
@@ -1293,9 +1297,17 @@ protected DocumentProjection getProjection() {
         DocumentProjection projection = new DocumentProjection(this.isIncludeGroupingContext(), this.isReducedResponse(), isTrackSizes());
         
         if (this.useWhiteListedFields) {
+            // make sure we include any fields being matched in the limit fields mechanism
+            if (!this.matchingFieldSets.isEmpty()) {
+                this.whiteListedFields.addAll(this.matchingFieldSets.stream().flatMap(s -> s.stream()).collect(Collectors.toList()));
```

Comment:
You do the following 3 times - consider encapsulating it into a protected method?
```
this.matchingFieldSets.stream().flatMap(s -> s.stream()).collect(Collectors.toList())
```

Reply:
done